wallet/wallet-cli/src/wallet_init.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common::chain::ChainConfig;
+use dialoguer::theme::ColorfulTheme;
+use wallet::Wallet;
+
+use crate::{
+    cli_println, errors::WalletCliError, helpers::select_helper, output::OutputContext, DefWallet,
+};
+
+#[derive(Clone, Copy)]
+enum ImportMnemonic {
+    Generate,
+    Import,
+    Cancel,
+}
+
+impl From<ImportMnemonic> for &str {
+    fn from(val: ImportMnemonic) -> Self {
+        match val {
+            ImportMnemonic::Generate => "Generate new mnemonic",
+            ImportMnemonic::Import => "Import mnemonic",
+            ImportMnemonic::Cancel => "Cancel",
+        }
+    }
+}
+
+pub fn new_wallet(
```

Comment:
This function, and the whole file, to be honest, is a good example of something that should go to that new crate.

Let's call `wallet-controller`.

Reply:
> This function, and the whole file, to be honest, is a good example of something that should go to that new crate.

I made small changes there and now `wallet_init.rs` has only CLI related functionality. 

> Let's call wallet-controller.

Do we really need another crate? Can't we use the root `wallet` crate for this?

Reply:
I do prefer creating a separate crate because using the wallet is not like using the wallet primitives. Some parametrization is being done, which is application-specific, which should be separate.